exec/java-exec/src/main/java/org/apache/drill/exec/server/options/TypeValidators.java

```diff
@@ -27,6 +27,21 @@
 import static com.google.common.base.Preconditions.checkArgument;
 
 public class TypeValidators {
+
+  /** Interface implemented by option validators which depend on other options
+   *  in order to perform validation. e.g. MIN/MAX option validators might be
+   *  dependent if MIN should always be less than MAX.
+   *
+   */
+  public interface DependentTypeValidators
+  {
+    /* Interface method requires providing an OptionManager which can
+     * be used to read option values of dependencies. As an example look at
+     * MinRangeDoubleValidator/MaxRangeDoubleValidator
+     */
+    public void validate(OptionValue v, BaseOptionManager manager);
```

Comment:
Would replacing `validate(OptionValue)` with `validate(OptionValue, OptionManager)` in the `OptionValidator` interface work?